File: src/main/java/ai/aimachineserver/domain/games/Game.java
package ai.aimachineserver.domain.games;

import org.springframework.web.socket.WebSocketSession;

import java.util.List;

public interface Game {
    void onPlayerJoinedGame(WebSocketSession session);

    void onDisconnect(WebSocketSession session);

    void onFieldClicked(int rowIndex, int colIndex);

    void broadcastMessage(KeyValue... keyValues);

    List<WebSocketSession> getAllPlayerSessions();
}

File: src/main/kotlin/ai/aimachineserver/domain/games/Game.kt
package ai.aimachineserver.domain.games

import org.json.JSONObject
import org.springframework.web.socket.WebSocketSession

interface Game {
    fun onPlayerJoinedGame(session: WebSocketSession)
    fun onDisconnect(session: WebSocketSession)
    fun broadcastMessage(eventObject: JSONObject)
    fun onFieldClicked(rowIndex: Int, colIndex: Int)

    fun getPlayerSessions(): List<WebSocketSession>
}
